keystrokes/src/components/Character/CharacterList.js

```diff
@@ -0,0 +1,151 @@
+import React, { useState, useEffect } from 'react';
+import '../../stylesheets/character.css';
+import styleGuide from '../../stylesheets/style';
+import { ref } from '../../config/constants';
+import { List, ListItem, Typography, Divider } from '@material-ui/core';
+import CharacterCreationModal from './CharacterCreationModal';
+import PropTypes from 'prop-types';
+
+CharacterList.propTypes = {
+	userInfo: PropTypes.object.isRequired,
+	userPath: PropTypes.string.isRequired,
+};
+
+function CharacterList({ userInfo, userPath }) {
+	const [characters, setCharacters] = useState({});
+	const [currentCharacter, setCurrentCharacter] = useState('');
+	const [open, setOpen] = useState(false);
+	const classes = styleGuide();
+
+	useEffect(() => {
+		setCharacters(userInfo.characters || {});
```

Comment:
This looks like a duplication of state to me. userInfo is stored at least one component up and i can't see characters changing specific to this file.

Reply:
`setCharacters({ ...characters, [idKey]: { name: name } }); ` I think just for this scenario, though maybe firebase has something with its data objects i dont know about.